src/main/java/com/minecolonies/coremod/network/messages/TransferItemsRequestMessage.java

```diff
@@ -113,41 +113,43 @@ public void messageOnServerThread(final TransferItemsRequestMessage message, fin
         }
 
         final Item item = message.itemStack.getItem();
-        final int amountToTake = Math.min(message.quantity, InventoryUtils.getItemCountInInventory(player.inventory, item, message.itemStack.getItemDamage()));
+        final int amountToTake = Math.min(message.quantity, InventoryUtils.getItemCountInItemHandler(new InvWrapper(player.inventory), item, message.itemStack.getItemDamage()));
         final ItemStack itemStackToTake = new ItemStack(item, amountToTake, message.itemStack.getItemDamage());
 
-        ItemStack remainingItemStack = InventoryUtils.setOverSizedStack(building.getTileEntity(), itemStackToTake);
+        ItemStack remainingItemStack = InventoryUtils.addItemStackToProviderWithResult(building.getTileEntity(), itemStackToTake);
 
-        //If we still have some to drop, let's try the additional chests now
-        if (remainingItemStack.getCount()>0)
+        if (!InventoryUtils.isItemStackEmpty(remainingItemStack))
         {
-            final World world = colony.getWorld();
-            for(final BlockPos pos : building.getAdditionalCountainers())
+            //If we still have some to drop, let's try the additional chests now
+            if (InventoryUtils.getItemStackSize(remainingItemStack) > 0)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1066)